cloud/awsprovider.go

```diff
@@ -20,20 +24,32 @@ import (
 	"github.com/aws/aws-sdk-go/aws/session"
 	"github.com/aws/aws-sdk-go/service/athena"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/jszwec/csvutil"
+
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
 // AWS represents an Amazon Provider
 type AWS struct {
 	Pricing          map[string]*AWSProductTerms
+	SpotPricing      map[string]*spotInfo
```

Comment:
For your consideration, `SpotPricingByInstanceId` or some godoc toward that end, since the two other maps in this struct (`Pricing` and `ValidPricingKeys`) are indexed by the "pricing tuple"

Reply:
Makes sense to me.